src/app/helpers/helperMethods.ts

```diff
@@ -1,37 +1,123 @@
 const path = require('path');
 const fs = require('fs');
+const hosts = [
+    "excel",
+    "onenote",
+    "outlook",
+    "powerpoint",
+    "project",
+    "word"
+];
 
 export namespace helperMethods {
-
-export function deleteFolderRecursively(projectFolder: string) 
-{
-    if(fs.existsSync(projectFolder))
+    export function deleteFolderRecursively(projectFolder: string) 
     {
-        fs.readdirSync(projectFolder).forEach(function(file,index){ 
-        var curPath = projectFolder + "/" + file; 
-        
-        if(fs.lstatSync(curPath).isDirectory())
+        if(fs.existsSync(projectFolder))
         {
-            deleteFolderRecursively(curPath);
+            fs.readdirSync(projectFolder).forEach(function(file,index){ 
+            var curPath = projectFolder + "/" + file; 
+            
+            if(fs.lstatSync(curPath).isDirectory())
+            {
+                deleteFolderRecursively(curPath);
+            }
+            else
+            {
+                fs.unlinkSync(curPath);
+            }
+        }); 
+        fs.rmdirSync(projectFolder); 
         }
-        else
+    }
+
+    export function doesProjectFolderExists(projectFolder: string)
+    {      
+    if (fs.existsSync(projectFolder))
         {
-            fs.unlinkSync(curPath);
+        if (fs.readdirSync(projectFolder).length > 0)
+        {          
+            return true;
+        }
         }
-    }); 
-    fs.rmdirSync(projectFolder); 
+        return false;
+    };
+
+    export function cleanupProjectFolder(projectFolder: string, host: string, typescript: boolean) {
+        return new Promise(async (resolve, reject) => {
+            try {
+                await _updateMoveProjectFiles(projectFolder, host, typescript);
```

Comment:
If you have two functions, one to move project files and the other to modify package.json, I would put those two into a function named modifyProjectForSingleHost() and have each named accordingly, e.g. modifyProjectFilesForSingleHost() and updatePackageJsonForSingleHost(). Makes it more clear what these functions do. 

Shouldn't there also be a check to skip doing this for Excel-Custom-Functions[-JS]?

Reply:
Yes, I will add check to skip ECF